### PR TITLE
feat(directory): configuration editor

### PR DESCRIPTION
**Pull request by jelmerveen**

### CODE REVIEW

File: apps/directory/src/components/partial-views/ApplicationHeaderPartialView.vue [hunk 1]
```diff
@@ -7,6 +7,14 @@
         </div>
       </div>
       <div class="col-4 d-flex justify-content-end">
+        <router-link
```
Comment (MaxPostema) [inline apps/directory/src/components/partial-views/ApplicationHeaderPartialView.vue:10]: I now have to click 3 times before the settings dialog opens. The second time there is a reload.

File: apps/directory/src/views/ConfigurationScreen.vue [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+                in with sufficient rights.</span
+              >
+            </div>
+          </div>
+
+          <div v-show="dirty" class="alert alert-warning m-0" role="alert">
+            <span>You have unsaved changes</span>
+          </div>
+        </div>
+        <small class="mt-4 float-right">To format your file press ctrl + f</small>
```
Comment (MaxPostema) [inline apps/directory/src/views/ConfigurationScreen.vue:160]: I think this only needs to be shown when the editor is also shown.

File: apps/directory/src/views/ConfigurationScreen.vue [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+      <landingpage-editor
+        :currentConfig="currentConfig"
+        @save="saveLandingpage"
+        @cancel="switchView('editor')"
+      />
+    </div>
+    <!-- standard button bar -->
+    <div v-if="currentView === 'ui'" class="row px-5 pb-5">
+      <div class="col pl-0">
+        <button class="btn btn-primary mr-3 save-button" @click="save">
```
Comment (MaxPostema) [inline apps/directory/src/views/ConfigurationScreen.vue:111]: The changes in ordering and names are saved correctly. But i don't see the changes reflected in the app.

File: apps/directory/src/views/ConfigurationScreen.vue [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+          Save configuration
+        </button>
+        <button v-if="dirty" class="btn btn-dark mr-3" @click="cancel">
+          Cancel
+        </button>
+
+        <button class="btn btn-outline-dark mr-3" @click="download">
+          Download config
+        </button>
+        <button class="btn btn-outline-dark" @click="upload">
```
Comment (MaxPostema) [inline apps/directory/src/views/ConfigurationScreen.vue:121]: Uploading a config crashes my browser ( chrome )

File: apps/directory/src/components/configuration/FilterConfigUI.vue [hunk 5]
```diff
@@ -0,0 +1,156 @@
+<template>
+  <div>
+    <div class="mb-3">
+      <h3>Filters</h3>
+      <div class="d-flex justify-content-between flex-wrap">
+        <small class="d-inline-block"
+          >Rearrange the filters by dragging and dropping. Unchecking a filter
+          means it will be shown under <i>More filters</i> by default.</small
+        >
+        <button @click="emitAdd" class="btn btn-info mt-2">
```
Comment (MaxPostema) [inline apps/directory/src/components/configuration/FilterConfigUI.vue:10]: This button does nothing.

File: apps/directory/src/components/configuration/FilterEditor.vue [hunk 6]
```diff
@@ -0,0 +1,97 @@
+<template>
+  <div @keyup.ctrl.f="format">
+    <div ref="filter-editor" class="filter-editor" @keyup="dirty = true"></div>
+
+    <button class="btn btn-info mt-3" :disabled="!dirty" @click="apply">
+      Apply changes to {{ title }}
+    </button>
+    <button class="btn btn-danger ml-3 mt-3" @click="deleteFilter">
```
Comment (MaxPostema) [inline apps/directory/src/components/configuration/FilterEditor.vue:8]: Delete action takes long but will not delete the filter.

File: apps/directory/src/components/configuration/FilterConfigUI.vue [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+          @click="activeFilter = element.facetTitle"
+          class="list-group-item d-flex"
+          :class="{ editing: element.index === filterIndex }"
+        >
+          {{ element.index + 1 }}. {{ element.label || element.facetTitle }}
+          <small class="ml-auto" v-if="element.builtIn"
+            >Rearranging this has no effect in the application.</small
+          >
+          <label v-if="!element.builtIn" class="ml-auto"
+            ><input type="checkbox" @change="sync"
```
Comment (MaxPostema) [inline apps/directory/src/components/configuration/FilterConfigUI.vue:35]: I don't understand what i can do with the checkboxes

**Review by MaxPostema — CHANGES_REQUESTED** [review]
Layout: i like the look, may need some small nudges. Like the save configuration button aligned with the filters box. And maybe the top of the editor with the top of the filters box.

File: apps/directory/src/components/partial-views/ApplicationHeaderPartialView.vue (hunk 1 above)
Comment (jelmerveen) [reply apps/directory/src/components/partial-views/ApplicationHeaderPartialView.vue:10]: somehow the lazy loading doesnt work for this screen. So removed that function. now one click again! (when logged in ;) )

File: apps/directory/src/views/ConfigurationScreen.vue (hunk 2 above)
Comment (jelmerveen) [reply apps/directory/src/views/ConfigurationScreen.vue]: done!

File: apps/directory/src/components/configuration/FilterConfigUI.vue (hunk 7 above)
Comment (jelmerveen) [reply apps/directory/src/components/configuration/FilterConfigUI.vue:35]: but now the checkboxes are wired in :) that actually will do something now

Comment (jelmerveen) [reply apps/directory/src/components/configuration/FilterConfigUI.vue]: the show more feature was dropped in previous version. so renamed the text to just hide

Comment (jelmerveen) [reply apps/directory/src/components/configuration/FilterConfigUI.vue]: also fixed layout as requested

File: apps/directory/src/views/ConfigurationScreen.vue (hunk 4 above)
Comment (jelmerveen) [reply apps/directory/src/views/ConfigurationScreen.vue]: fixed now. download/upload can only be done now in json tab

File: apps/directory/src/components/configuration/FilterConfigUI.vue (hunk 5 above)
Comment (jelmerveen) [reply apps/directory/src/components/configuration/FilterConfigUI.vue]: fixed

**Comment by jelmerveen** [comment]
also fixed the card layout as suggested

**Review by MaxPostema — APPROVED** [review]

**Pull request merged**